File: project_euler/Problem076/Problem076.py
class Problem076:
    def solution(self) -> None:
        target = 100
        ways = [0 for _ in range(target + 1)]
        ways[0] = 1
        for i in range(1, target):
            for j in range(i, target + 1):
                ways[j] += ways[j - i]
        print(ways[target])

File: project_euler/Problem076/Problem076.java
package project_euler;

import java.util.Arrays;

public class Problem076 {
    public void solution() {
        int target = 100;
        int[] ways = new int[target + 1];
        Arrays.fill(ways, 0);
        ways[0] = 1;
        for (int i = 1; i < target; ++i) {
            for (int j = i; j <= target; ++j) {
                ways[j] += ways[j - i];
            }
        }
        System.out.println(ways[target]);
    }
}
